### DISCUSSION TITLE
开发大大你好，目前框架貌似只能直接回复发送者的消息，请问会考虑转发消息吗？

### DISCUSSION THREAD

**Discussion by GeekSheng** [Hoshikawa-Kaguya/Sora#14] [Q&A] [answered]
开发大大你好，目前框架貌似只能直接回复发送者的消息，请问会考虑转发消息吗，比如我想把群内的闪图转发到我的个人qq号上？

**Comment by Yukari316** [comment, marked as answer]
转发的功能是需要你自己去编写逻辑的，框架只会提供访问`go-cqhttp`的API接口

如果是想要转发闪照，由于闪照无法获取图片url，只能使用缓存文件转发
参考以下代码：
```csharp
var img = eventArgs.Message.GetAllImage();
await eventArgs.SourceGroup.SendGroupMessage(CQCode.CQImage(img[0].ImgFile));
```
这只是简单的示例，实际的业务逻辑需要你自己去完成（毕竟我也不知道你们要拿来干什么.jpg

**Reply by d3ara1n** [reply]
闪照在gocqhttp中只会提供imageid，而imageid可以获取url，换句话说，闪照和图片是一个东西。可以在api层直接等效化闪照和图片。

**Reply by Yukari316** [reply]
试过了，闪照的`imageid`用`get_image`是拿不到url的
![image](https://user-images.githubusercontent.com/7535224/114551388-e7537a80-9c95-11eb-9bbe-0263c7d3b5b6.png)

**Reply by Yukari316** [reply]
补充一点， `file` 字段里虽说有文件路径，但这个文件是不存在的

**Reply by d3ara1n** [reply]
imageid2url的逻辑要自己实现，这个问题之前mirai群里讨论过，腾讯官方给了接口不需要gocqhttp所以文档里是没有的。

**Reply by d3ara1n** [reply]
> imageid2url的逻辑要自己实现，这个问题之前mirai群里讨论过，腾讯官方给了接口不需要gocqhttp所以文档里是没有的。

刚去试了一下目前方法可行。
```csharp
FlashImage.Open = (id) => HttpReadAsStream($"http://gchat.qpic.cn/gchatpic_new/0/0-0-{id.ToUpper()}/0?term=0");
```
这个与腾讯qq服务耦合，可以放在应用层实现而不是由SDK实现。

**Reply by Yukari316** [reply]
可以做到框架里但是确实没什么必要就是了
